sdk/azcore/runtime/pager.go

```diff
@@ -0,0 +1,95 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package runtime
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"reflect"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/shared"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+// PageProcessor contains the required data for constructing a Pager.
+type PageProcessor[T any] struct {
+	// Do sends the request to fetch the next page.
+	Do func(*policy.Request) (*http.Response, error)
+
+	// More returns a boolean indicating if there are more pages to fetch.
+	// It uses the provided page to make the determination.
+	More func(T) bool
+
+	// Requester creates the request to fetch the first page.
+	Requester func(context.Context) (*policy.Request, error)
```

Comment:
Not that I want to redesign the API here, but do we need two methods for fetching pages? Naively, it seems `Requester` would be a straightforward special case for `Advancer`.

Reply:
That's a good point.  Maybe we can fold this into a func named `Fetcher`, and when requesting the first page, the `T` is its zero-value?  I don't think there could ever be a case where that's ambiguous (or we just make it pointer-to-T).

Reply:
I went with the pointer-to-T as it simplifies the condition and removed the need to use reflection.